.github/workflows/ci.yml

```
@@ -0,0 +1,288 @@
+        with:
+          fetch-depth: 0
+      
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
+          cache: true
+      
+      - name: Check if should continue
+        id: check
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2193:warning:2:49: The arguments to this comparison can never be equal. Make sure your syntax is correct

(shellcheck)